### PR TITLE
Port check_plugin workflow to lightweight script

### PR DESCRIPTION
**Pull request by peterzhuamazon**
This PR is to port the check_plugin.yml workflow to a script for lightweight access by other workflows.

### Test Cases:
1. Success case: 
https://github.com/opendistro-for-elasticsearch/opendistro-build/actions/runs/119475111
2. Failure case:
https://github.com/opendistro-for-elasticsearch/opendistro-build/actions/runs/119473418

### CODE REVIEW

File: .github/scripts/check_plugin.sh [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+S3_DIR_kibana="downloads/kibana-plugins"
+TSCRIPT_NEWLINE="%0D%0A"
+RUN_STATUS=0 # 0 is success, 1 is failure
+PLUGIN_TYPES=$1
+ODFE_VERSION=$2
+#CHIME_ALERT=$3
+#EMAIL_ALERT=$4
+
+# This script allows users to manually assign parameters
+#if [ "$#" -gt 0 ] && [ "$#" -ne 4 ]
```
Comment (weicongs-amazon) [inline .github/scripts/check_plugin.sh:19]: remove these if it's useless

File: .github/scripts/check_plugin.sh [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+if [ "$#" -gt 0 ] && [ "$#" -ne 2 ]
+then
+  #echo "ERROR: Please assign 0 or 4 parameters when running this script"
+  #echo "Example: $0 \"rpm,kibana\" \"1.7.0\" \"false\" \"true\""
+  #echo "This will check rpm & kibana plugins, NOT send Chime MSG, send Email MSG"
```
Comment (weicongs-amazon) [inline .github/scripts/check_plugin.sh:24]: the same as above

File: .github/scripts/check_plugin.sh [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+#echo "#######################################"
+#
+#echo "#######################################"
+#echo "EMAIL_ALERT is: $EMAIL_ALERT"
+#if [ -z "$EMAIL_ALERT" ]
+#then
+#  EMAIL_ALERT=true
+#  echo "Use default EMAIL_ALERT: $EMAIL_ALERT"
+#fi
+#echo "#######################################"
```
Comment (weicongs-amazon) [inline .github/scripts/check_plugin.sh:69]: the same as above

File: .github/workflows/check_plugin_old.yml [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        uses: dawidd6/action-send-mail@master
+        with:
+          server_address: smtp.gmail.com
+          server_port: 465
+          username: ${{secrets.MAIL_USERNAME}}
+          password: ${{secrets.MAIL_PASSWORD}}
+          subject: Opendistro for Elasticsearch Build - Daily Run (Plugin Status)
+          # Read file contents as body:
+          body: file://message.md
+          to: sngri@amazon.com,odfe-distribution-build@amazon.com
```
Comment (gaiksaya) [inline .github/workflows/check_plugin_old.yml:255]: You can remove Rishabh's email id from here

File: .github/scripts/check_plugin.sh (hunk 1 above)
Comment (peterzhuamazon) [reply .github/scripts/check_plugin.sh]: Resolved.

File: .github/scripts/check_plugin.sh (hunk 2 above)
Comment (peterzhuamazon) [reply .github/scripts/check_plugin.sh]: Resolved.

File: .github/scripts/check_plugin.sh (hunk 3 above)
Comment (peterzhuamazon) [reply .github/scripts/check_plugin.sh]: Resolved.

File: .github/workflows/check_plugin_old.yml (hunk 4 above)
Comment (peterzhuamazon) [reply .github/workflows/check_plugin_old.yml:255]: Resolved.

**Pull request merged**